backend-es/src/PureScript/Backend/Optimizer/Codegen/EcmaScript/Convert.purs

```diff
@@ -621,19 +622,19 @@ codegenObjectElement env (Prop p1 expr) =
       EsObjectField p1 other
 
 codegenCtor :: CodegenEnv -> ModuleName -> ConstructorType -> ProperName -> Ident -> Array EsExpr -> EsExpr
-codegenCtor env@(CodegenEnv { currentModule, options }) mod ct name tag values = case ct of
+codegenCtor env@(CodegenEnv { currentModule, options }) mod ct name tagIdent values = case ct of
   SumType -> do
-    let ctorMeta = lookupCtorMeta env (Qualified (Just mod) tag)
-    build $ EsCall ctorName $ Array.cons (EsArrayValue (codegenTag options tag ctorMeta)) ctorValues
+    let { tag } = lookupCtorInfo env (Qualified (Just mod) tagIdent)
+    build $ EsCall ctorName $ Array.cons (EsArrayValue (codegenTag options tagIdent tag)) ctorValues
   ProductType ->
     build $ EsCall ctorName ctorValues
   where
   ctorName = build $ EsIdent $ Qualified ctorModule $ asCtorIdent name
   ctorModule = if mod == currentModule then Nothing else Just mod
   ctorValues = EsArrayValue <$> values
 
-codegenTag :: CodegenOptions -> Ident -> CtorMeta -> EsExpr
-codegenTag opts (Ident ctor) { tag }
+codegenTag :: CodegenOptions -> Ident -> Int -> EsExpr
```

Comment:
Was there a reason for changing this?

Reply:
This was changed because `lookupCtorMeta` was changed to 
```purescript
lookupCtorInfo :: CodegenEnv -> Qualified Ident -> { tag :: Int, size :: Int }
```
which doesn't return `CtorMeta` anymore but `tag` from `CtorMeta` and `size` from `DataTypeMeta`, which is required for the optimization. Therefor the callers of `codegenTag` which use `lookupCtorInfo` don't have `CtorMeta` available anymore, but only `tag`.

Another possibility would be of course to change `lookupCtorInfo` to:
```purescript
lookupCtorInfo :: CodegenEnv -> Qualified Ident -> { meta :: CtorMeta, size :: Int }
```
then `codegenTag` could remain unchanged.

Reply:
I think I'd prefer this.

Reply:
Done.